Library/Homebrew/cmd/deps.rb

```diff
@@ -14,16 +14,19 @@ def deps
 
     if mode.installed? && mode.tree?
       puts_deps_tree Formula.installed
-    elsif mode.installed?
-      puts_deps Formula.installed
     elsif mode.all?
       puts_deps Formula
     elsif mode.tree?
       raise FormulaUnspecifiedError if ARGV.named.empty?
       puts_deps_tree ARGV.formulae
     else
-      raise FormulaUnspecifiedError if ARGV.named.empty?
+      if ARGV.named.empty?
+        raise FormulaUnspecifiedError unless mode.installed?
+        puts_deps Formula.installed
+        return
```

Comment:
Maybe
```ruby
elsif ARGV.named.empty?
  ...
else
  ...
end
```
instead of
```ruby
else
  if ARGV.named.empty?
    ...
    return
  end
  ...
end
```